lib/models/environment-variables.coffee

```diff
@@ -143,3 +143,126 @@ exports.remove = (id, callback) ->
 ####
 exports.isSystemVariable = (variable) ->
 	return /^RESIN_/.test(variable.name)
+
+exports.device = {}
+
+###*
+# device.getAll callback
+# @callback module:resin.models.environment-variables.device~getAllCallback
+# @param {(Error|null)} error - error
+# @param {EnvironmentVariable[]} environmentVariables - environment variables
+###
+
+###*
+# @summary Get all device environment variables
+# @public
+# @function
+#
+# @param {(String|Number)} deviceId - device id
+# @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
+#
+# @example
+#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#		throw error if error?
+#		console.log(environmentVariables)
+###
+exports.device.getAll = (deviceId, callback) ->
+	return pine.get
+		resource: 'device_environment_variable'
+		options:
+			filter:
+				device: deviceId
+			expand: 'device'
+			orderby: 'env_var_name asc'
+	.tap (environmentVariables) ->
+		if _.isEmpty(environmentVariables)
+			throw new errors.ResinNotAny('device environment variables')
```

Comment:
I don't know if this is the standard way of handling an empty list in the SDK, but I'd expect to just get an empty array returned since I'm asking for all the ones that exist - whereas if I was asking for one specific instance then I'd expect an error for it to be missing (since I'm looking for that particular one by name/id)